components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -256,39 +256,22 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
     ): Promise<void> {
         await super.mayStartWorkspace(ctx, user, runningInstances);
 
-        // TODO(at) replace the naive implementation based on usage service
-        // with a proper call check against the upcoming invoice.
-        // For now this should just enable the work on fronend.
-        if (await this.isUsageBasedFeatureFlagEnabled(user)) {
-            // dummy implementation to test frontend bits
-            const attributionId = await this.userService.getWorkspaceUsageAttributionId(user);
-            const costCenter = !!attributionId && (await this.costCenterDB.findById(attributionId));
-            if (costCenter) {
-                const allSessions = await this.listBilledUsage(ctx, {
-                    attributionId,
-                    startedTimeOrder: SortOrder.Descending,
-                });
-                const totalUsage = allSessions.map((s) => s.credits).reduce((a, b) => a + b, 0);
-
-                if (totalUsage >= costCenter.spendingLimit) {
-                    throw new ResponseError(
-                        ErrorCodes.PAYMENT_SPENDING_LIMIT_REACHED,
-                        "Increase spending limit and try again.",
-                        {
-                            attributionId: user.usageAttributionId,
-                        },
-                    );
-                }
-            }
-        }
-
         const result = await this.entitlementService.mayStartWorkspace(user, new Date(), runningInstances);
         if (!result.enoughCredits) {
             throw new ResponseError(
                 ErrorCodes.NOT_ENOUGH_CREDIT,
                 `Not enough monthly workspace hours. Please upgrade your account to get more hours for your workspaces.`,
             );
         }
+        if (result.spendingLimitReachedOnCostCenter) {
```

Comment:
need to verify this, but it seems to be wrong for EntitlementServiceUBP.mayStartWorkspace as it wont return `.enoughCredits` in result, thus the error mode will be mixed up. 

I suppose we need to get the signals aligned (enoughCredits -> notEnoughCredits, for instance) so that the pattern is clear.

Reply:
<img width="757" alt="Screen Shot 2022-08-10 at 09 44 45" src="https://user-images.githubusercontent.com/914497/183844446-50093e6e-64e9-4e6c-b8df-d57a72a909b6.png">

🎯  finally, was able to actually run the stuff and verified that's it's an issue indeed. 
going to apply the refactoring mentioned above, just because it will solve the problem in a readable way.